src/t_zset.c

```diff
@@ -327,10 +347,8 @@ static zskiplistNode *zslUpdateScore(zskiplist *zsl, zskiplistNode *node, double
 
     zslDeleteNode(zsl, node, update);
     /* update pointer inside hashtable with new node */
-    zskiplistNode *new_node = zslInsert(zsl, newscore, node->ele);
-    /* We reused the old node->ele SDS string, free the node now
-     * since zslInsert created a new node */
-    node->ele = NULL;
+    sds ele = zslGetNodeElement(node);
+    zskiplistNode *new_node = zslInsert(zsl, newscore, ele);
```

Comment:
question: can't we just use zslInsertNode with the already existing node?
maybe now that we will have to relocate potentially large strings it will be proven more useful?